src/electric/adapter.ts

```diff
@@ -1,19 +1,22 @@
 import { AnyDatabase } from '../drivers/index'
 import { QualifiedTablename } from '../util/tablename'
-import { BindParams, Query, Row } from '../util/types'
+import { Row, Statement } from '../util/types'
 
 // A `DatabaseAdapter` adapts a database client to provide the
 // normalised interface defined here.
 export interface DatabaseAdapter {
   db: AnyDatabase
 
-  // Runs the provided sql.
-  run(sql: string): Promise<void>
+  // Runs the provided sql statement
+  run(statement: Statement): Promise<void>
+
+  // Runs the provided sql as a transaction
+  runTransaction(...statements: Statement[]): Promise<void>
 
   // Query the database.
-  query(query: Query, bindParams?: BindParams): Promise<Row[]>
+  query(statement: Statement): Promise<Row[]>
```

Comment:
There's a nuance here where some drivers (in future) may need to handle query objects. For example, when using TypeORM. Perhaps we can cross that bridge when we get to it though. Or make this `Statement | Query`.

Reply:
Yeah. Deliberately removed overloads of the interface. We extend it as we go.